torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -901,25 +948,44 @@ class TritonHOPifier:
     TritonHOPifier is an abstract class that can be overriden by its subclasses.
     """
 
-    def raise_unsupported(self, msg):
+    def raise_unsupported(self, msg: str) -> None:
         raise NotImplementedError("abstract method")
 
-    def is_callable(self, maybe_callable):
+    def is_callable(self, maybe_callable: Any) -> bool:
         raise NotImplementedError("abstract method")
 
-    def get_value(self, val):
+    def get_value(self, val: Any) -> Any:
         raise NotImplementedError("abstract method")
 
-    def call_grid(self, grid, meta, tx):
+    def call_grid(
+        self,
+        grid: "Union[TritonGridCallableType, VariableTracker]",
+        meta: "Union[TritonMetaParamsType, VariableTracker]",
+        tx: Optional["InstructionTranslator"],
+    ) -> Union[Tuple[Union[int, sympy.Expr, SymInt], ...], Tuple["Proxy", ...]]:
         raise NotImplementedError("abstract method")
 
-    def call_HOP(self, variable, grids, combined_args, tx):
+    def call_HOP(
+        self,
+        variable: "Union[TraceableTritonKernelWrapper, TritonKernelVariable]",
+        grids: List["TritonGridTupleType"],
+        combined_args: Dict[str, Any],
+        tx: Optional["InstructionTranslator"],
+    ) -> Optional["ConstantVariable"]:
```

Comment:
In the non-Dynamo case this shouldn't be ConstantVariable

Reply:
But this is the base class - `call_HOP` can _either_ return None, or ConstantVariable - hence `Optional[ConstantVariable]`. Does that sound right or am I missing something?

Reply:
Sorry, I didn't see the Optional part, you're right